### ISSUE TITLE
PokemonOptimizer decreases gained stardust count

### ISSUE THREAD

**Issue by cmezh** [PokemonGoF/PokemonGo-Bot#5350]
After pokemon upgrading gained stardust count decreases to nevative values

```
[2016-09-09 22:22:18] Uptime : 5:24:20 | 13.44km walked | Earned 10,951 Stardust | Level 28 (73,880 / 300,000, 24%) | +67,895 XP | 12,560 XP/h | Visited 239 stops | Caught 80 pokemon | Evolved 79 pokemon
[2016-09-09 22:22:18] Transferring 66 Pokemon
[2016-09-09 22:22:19] Exchanged Beedrill [IV 0.78] [CP 633] [123 candies]
[2016-09-09 22:22:22] Exchanged Pidgeotto [IV 0.84] [CP 531] [8 candies]
[2016-09-09 22:22:26] Exchanged Pidgeotto [IV 0.8] [CP 824] [9 candies]
[2016-09-09 22:22:30] Exchanged Pidgeotto [IV 0.76] [CP 667] [10 candies]
[2016-09-09 22:22:35] Exchanged Pidgeotto [IV 0.76] [CP 327] [11 candies]
[2016-09-09 22:22:41] Exchanged Pidgeotto [IV 0.73] [CP 682] [12 candies]
[2016-09-09 22:22:44] Exchanged Pidgeotto [IV 0.67] [CP 150] [13 candies]
[2016-09-09 22:22:49] Exchanged Pidgeotto [IV 0.67] [CP 148] [14 candies]
[2016-09-09 22:22:53] Exchanged Pidgeotto [IV 0.62] [CP 633] [15 candies]
[2016-09-09 22:22:58] Exchanged Pidgeotto [IV 0.62] [CP 48] [16 candies]
[2016-09-09 22:23:04] Exchanged Pidgeotto [IV 0.6] [CP 85] [17 candies]
[2016-09-09 22:23:08] Exchanged Pidgeotto [IV 0.58] [CP 720] [18 candies]
[2016-09-09 22:23:12] Exchanged Pidgeotto [IV 0.56] [CP 521] [19 candies]
[2016-09-09 22:23:16] Exchanged Pidgeotto [IV 0.56] [CP 506] [20 candies]
[2016-09-09 22:23:21] Exchanged Pidgeotto [IV 0.56] [CP 499] [21 candies]
[2016-09-09 22:23:25] Exchanged Pidgeotto [IV 0.56] [CP 112] [22 candies]
[2016-09-09 22:23:29] Exchanged Pidgeotto [IV 0.53] [CP 722] [23 candies]
[2016-09-09 22:23:34] Exchanged Pidgeotto [IV 0.53] [CP 495] [24 candies]
[2016-09-09 22:23:38] Exchanged Pidgeotto [IV 0.53] [CP 466] [25 candies]
[2016-09-09 22:23:42] Exchanged Pidgeotto [IV 0.53] [CP 280] [26 candies]
[2016-09-09 22:23:46] Exchanged Pidgeotto [IV 0.51] [CP 547] [27 candies]
[2016-09-09 22:23:50] Exchanged Pidgeotto [IV 0.51] [CP 497] [28 candies]
[2016-09-09 22:23:55] Exchanged Pidgeotto [IV 0.51] [CP 462] [29 candies]
[2016-09-09 22:23:59] Exchanged Pidgeotto [IV 0.51] [CP 176] [30 candies]
[2016-09-09 22:24:03] Exchanged Pidgeotto [IV 0.49] [CP 762] [31 candies]
[2016-09-09 22:24:08] Exchanged Pidgeotto [IV 0.49] [CP 644] [32 candies]
[2016-09-09 22:24:13] Exchanged Pidgeotto [IV 0.47] [CP 635] [33 candies]
[2016-09-09 22:24:17] Exchanged Pidgeotto [IV 0.47] [CP 517] [34 candies]
[2016-09-09 22:24:22] Exchanged Pidgeotto [IV 0.44] [CP 737] [35 candies]
[2016-09-09 22:24:27] Exchanged Pidgeotto [IV 0.44] [CP 579] [36 candies]
[2016-09-09 22:24:32] Exchanged Pidgeotto [IV 0.42] [CP 516] [37 candies]
[2016-09-09 22:24:36] Exchanged Raticate [IV 0.87] [CP 876] [3 candies]
[2016-09-09 22:24:41] Exchanged Raticate [IV 0.8] [CP 592] [4 candies]
[2016-09-09 22:24:46] Exchanged Raticate [IV 0.73] [CP 699] [5 candies]
[2016-09-09 22:24:51] Exchanged Raticate [IV 0.73] [CP 426] [6 candies]
[2016-09-09 22:24:56] Exchanged Raticate [IV 0.73] [CP 390] [7 candies]
[2016-09-09 22:25:00] Exchanged Raticate [IV 0.71] [CP 463] [8 candies]
[2016-09-09 22:25:04] Exchanged Raticate [IV 0.71] [CP 425] [9 candies]
[2016-09-09 22:25:09] Exchanged Raticate [IV 0.71] [CP 19] [10 candies]
[2016-09-09 22:25:14] Exchanged Raticate [IV 0.69] [CP 968] [11 candies]
[2016-09-09 22:25:18] Exchanged Raticate [IV 0.69] [CP 508] [12 candies]
[2016-09-09 22:25:23] Exchanged Raticate [IV 0.69] [CP 499] [13 candies]
[2016-09-09 22:25:28] Exchanged Raticate [IV 0.67] [CP 1003] [14 candies]
[2016-09-09 22:25:33] Exchanged Raticate [IV 0.67] [CP 771] [15 candies]
[2016-09-09 22:25:37] Exchanged Raticate [IV 0.67] [CP 346] [16 candies]
[2016-09-09 22:25:42] Exchanged Raticate [IV 0.64] [CP 961] [17 candies]
[2016-09-09 22:25:46] Exchanged Raticate [IV 0.58] [CP 1051] [18 candies]
[2016-09-09 22:25:50] Exchanged Fearow [IV 0.87] [CP 971] [34 candies]
[2016-09-09 22:25:54] Exchanged Fearow [IV 0.87] [CP 127] [35 candies]
[2016-09-09 22:26:00] Exchanged Fearow [IV 0.78] [CP 769] [36 candies]
[2016-09-09 22:26:04] Exchanged Fearow [IV 0.73] [CP 954] [37 candies]
[2016-09-09 22:26:07] Exchanged Fearow [IV 0.69] [CP 424] [38 candies]
[2016-09-09 22:26:13] Exchanged Fearow [IV 0.62] [CP 887] [39 candies]
[2016-09-09 22:26:18] Exchanged Gengar [IV 0.71] [CP 898] [74 candies]
[2016-09-09 22:26:23] Exchanged Hypno [IV 0.82] [CP 1329] [47 candies]
[2016-09-09 22:26:28] Exchanged Hypno [IV 0.82] [CP 1075] [48 candies]
[2016-09-09 22:26:32] Exchanged Hypno [IV 0.8] [CP 1144] [49 candies]
[2016-09-09 22:26:37] Exchanged Hypno [IV 0.78] [CP 1540] [50 candies]
[2016-09-09 22:26:42] Exchanged Hypno [IV 0.78] [CP 897] [51 candies]
[2016-09-09 22:26:47] Exchanged Hypno [IV 0.78] [CP 784] [52 candies]
[2016-09-09 22:26:52] Exchanged Hypno [IV 0.76] [CP 283] [53 candies]
[2016-09-09 22:26:57] Exchanged Hypno [IV 0.73] [CP 1330] [54 candies]
[2016-09-09 22:27:03] Exchanged Hypno [IV 0.73] [CP 471] [55 candies]
[2016-09-09 22:27:06] Exchanged Hypno [IV 0.69] [CP 1423] [56 candies]
[2016-09-09 22:27:10] Exchanged Flareon [IV 0.56] [CP 1770] [8 candies]
[2016-09-09 22:27:15] Exchanged Flareon [IV 0.49] [CP 406] [9 candies]
[2016-09-09 22:27:19] Pokemon Bag: 179/250
[2016-09-09 22:27:20] Waiting for more Pokemon to evolve with lucky egg: 0/80
[2016-09-09 22:27:20] Upgrading 1 Pokemon [72281 stardust]
[2016-09-09 22:27:20] Upgraded Beedrill [IV 0.96] [CP 817] [121 candies] [69781 stardust]
[2016-09-09 22:27:25] Upgraded Beedrill [IV 0.96] [CP 817] [119 candies] [67281 stardust]
[2016-09-09 22:27:29] Upgraded Beedrill [IV 0.96] [CP 817] [117 candies] [64781 stardust]
[2016-09-09 22:27:34] Upgraded Beedrill [IV 0.96] [CP 817] [114 candies] [61781 stardust]
[2016-09-09 22:27:39] Upgraded Beedrill [IV 0.96] [CP 817] [111 candies] [58281 stardust]
[2016-09-09 22:27:43] Upgraded Beedrill [IV 0.96] [CP 817] [108 candies] [55281 stardust]
[2016-09-09 22:27:47] Upgraded Beedrill [IV 0.96] [CP 817] [105 candies] [52281 stardust]
[2016-09-09 22:27:52] Upgraded Beedrill [IV 0.96] [CP 817] [102 candies] [48781 stardust]
[2016-09-09 22:27:56] Upgraded Beedrill [IV 0.96] [CP 817] [99 candies] [45281 stardust]
[2016-09-09 22:27:59] Upgraded Beedrill [IV 0.96] [CP 817] [96 candies] [41281 stardust]
[2016-09-09 22:28:04] Upgraded Beedrill [IV 0.96] [CP 817] [93 candies] [37781 stardust]
[2016-09-09 22:28:09] Upgraded Beedrill [IV 0.96] [CP 817] [89 candies] [33281 stardust]
[2016-09-09 22:28:14] Upgraded Beedrill [IV 0.96] [CP 817] [85 candies] [29281 stardust]
[2016-09-09 22:28:18] Upgraded Beedrill [IV 0.96] [CP 817] [81 candies] [25281 stardust]
[2016-09-09 22:28:23] Upgraded Beedrill [IV 0.96] [CP 817] [77 candies] [21281 stardust]
[2016-09-09 22:28:27] Upgraded Beedrill [IV 0.96] [CP 817] [73 candies] [16281 stardust]
[2016-09-09 22:28:31] Upgraded Beedrill [IV 0.96] [CP 817] [69 candies] [11781 stardust]
[2016-09-09 22:28:35] Upgraded Beedrill [IV 0.96] [CP 817] [65 candies] [7281 stardust]
[2016-09-09 22:54:05] Uptime : 5:56:07 | 13.44km walked | Earned -54,049 Stardust | Level 28 (73,880 / 300,000, 24%) | +67,895 XP | 11,439 XP/h | Visited 239 stops | Caught 80 pokemon | Evolved 79 pokemon
```

**Comment by rawgni** [comment]
@cmezh What you gain today is -, since you only gain 10951 today and you use 72281 to evolve.

**Comment by Gobberwart** [comment]
@rawgni is correct. Upgrading costs stardust :)

**Comment by cmezh** [comment]
@rawgni You're correct. But I think it shouldn't count spent stardust, because it is just EARNED stardust :)
Otherwise total stardust info is enough (#5338)

**Comment by rawgni** [comment]
or we could have EARNED, USED, TOTAL 💃

**Comment by cmezh** [comment]
@rawgni Definetely good idea :+1:

**Comment by DBa2016** [comment]
"total" is already addressed by #5338

**Comment by Gobberwart** [comment]
Stale. Closing.
